src/nanoarrow/utils_test.cc

```diff
@@ -543,6 +543,36 @@ TEST(DecimalTest, DecimalRoundtripBitshiftTest) {
   ArrowBufferReset(&buffer);
 }
 
+// test case adapted from
+// https://github.com/apache/arrow/blob/main/go/arrow/float16/float16_test.go
+TEST(HalfFloatTest, FloatToHalfFloat) {
```

Comment:
We should specifically test that NaNs are preserved as well. It's a simple check, fortunately: https://github.com/apache/arrow/blob/6800be9331d88024bf550c77865a06c592a22699/go/arrow/float16/float16.go#L176